### PR TITLE
Add missing urls for Trusted* (and toJSON() entries)

### PR DESCRIPTION
**Pull request by teoli2003**
Added the mdn_url and spec_url to `TrustedHTML`, `TrustedScript`, `TrustedScriptURL`, `TrustedTypePolicy`, and `TrustedTypePolicyFactor`(both on the interface and the children).

Added the `TrustedHTML.toJSON()`, `TrustedScript.toJSON()`, and `TrustedScriptURL.toJSON()` compat data. They are only in Canary, so I put a note there.

### CODE REVIEW

**Comment by teoli2003** [comment]
For the record, these, including the `toJSON()` methods, have been added to MDN web docs in mdn/content#2978

File: api/TrustedHTML.json [hunk 1]
```diff
@@ -46,8 +48,60 @@
           "deprecated": false
         }
       },
+      "toJSON": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/TrustedHTML/toJSON",
```
Comment (foolip) [inline api/TrustedHTML.json:53]: Wow, https://developer.mozilla.org/docs/Web/API/TrustedHTML/toJSON actually exists :)

Commented on something at https://github.com/mdn/content/pull/2978#pullrequestreview-686087283...

File: api/TrustedHTML.json [hunk 2]
```diff
@@ -46,8 +48,60 @@
...
+          "spec_url": "https://w3c.github.io/webappsec-trusted-types/dist/spec/#dom-trustedhtml-tojson",
+          "support": {
+            "chrome": {
+              "version_added": false,
+              "notes": "Currently Chrome Canary only"
```
Comment (foolip) [inline api/TrustedHTML.json:58]: Very few things are enabled in Chrome Canary only in the same way that Firefox does, unless perhaps we count all of the "experimental web platform features" flag in this bucket.

Anyway, how did you determine this is in Canary only? Judging by https://mdn-bcd-collector.appspot.com/tests/api/TrustedHTML/toJSON it's already shipped to stable in Chrome 90, and that happened in this commit:
https://storage.googleapis.com/chromium-find-releases-static/0db.html#0dbce1408181f002902cba22fa275f658e8d9c18

**Review by foolip — CHANGES_REQUESTED** [review]

**Comment by 138166234** [comment]
#3060

1381662341mohamadsalmanpor

در تاریخ پنجشنبه ۱۷ ژوئن ۲۰۲۱،‏ ۱۴:۱۳ Philip Jägenstedt <
***@***.***> نوشت:

File: api/TrustedHTML.json (hunk 1 above)
Comment (teoli2003) [reply api/TrustedHTML.json:53]: We have a few toJSON and toString (stringifier) pages. Like for the iterable pages, they all have a close structure that make them fairly quick to create.

File: api/TrustedHTML.json (hunk 2 above)
Comment (teoli2003) [reply api/TrustedHTML.json:58]: I trusted @RachelAndrew comment in  mdn/content#297, and it could have changed in the meantime!
Should we put 90?

Comment (foolip) [reply api/TrustedHTML.json:58]: Yeah, I'd use Chrome 90 and then `npm run mirror` to update the rest.

Comment (teoli2003) [reply api/TrustedHTML.json]: I tried to do it. My first run wiith `npm run mirror`, so I hope I did well. Fix in next commit

File: api/TrustedHTML.json [hunk 3]
```diff
@@ -46,8 +48,59 @@
...
+          "spec_url": "https://w3c.github.io/webappsec-trusted-types/dist/spec/#dom-trustedhtml-tojson",
+          "support": {
+            "chrome": {
+              "version_added": "90"
+            },
+            "chrome_android": {
+              "version_added": "90"
+            },
+            "edge": {
+              "version_added": null
```
Comment (foolip) [inline api/TrustedHTML.json:63]:
```suggestion
              "version_added": "90"
```

Just mirroring Edge is typically safe for new APIs, but also confirmed by testing Edge 89 and 90 on https://mdn-bcd-collector.appspot.com/tests/api/TrustedHTML/toJSON.

File: api/TrustedScript.json [hunk 4]
```diff
@@ -46,8 +48,59 @@
...
+          "spec_url": "https://w3c.github.io/webappsec-trusted-types/dist/spec/#dom-trustedscript-tojson",
+          "support": {
+            "chrome": {
+              "version_added": "90"
+            },
+            "chrome_android": {
+              "version_added": "90"
+            },
+            "edge": {
+              "version_added": false
```
Comment (foolip) [inline api/TrustedScript.json:63]:
```suggestion
              "version_added": "90"
```

Confirmed using https://mdn-bcd-collector.appspot.com/tests/api/TrustedScript/toJSON

**Review by foolip — APPROVED** [review]
Missing some Edge data, but I'll apply my own suggestions and merge.

File: api/TrustedHTML.json (hunk 1 above)
Comment (foolip) [reply api/TrustedHTML.json:53]: IMHO, for them to be useful, they would have to say which attributes are actually serialized in the resulting JSON, since that's often defined by prose in specs and isn't really guessable from looking at https://developer.mozilla.org/en-US/docs/Web/API/TrustedHTML, for example.

File: api/TrustedHTML.json (hunk 2 above)
Comment (foolip) [reply api/TrustedHTML.json]: I didn't try running it myself, but `npm run mirror edge` should have mirrored the Edge data, but you have to run it once for each file you care about so it's very tedious. https://github.com/mdn/browser-compat-data/pull/10645 will make it less tedious.

Comment (teoli2003) [reply api/TrustedHTML.json]: I tried, but I think it mirrored it to false as it took IE as source. Maybe I did it wrong, though.

Comment (foolip) [reply api/TrustedHTML.json]: Ah right, https://github.com/mdn/browser-compat-data/pull/6582 hasn't been merged yet :)
